jinja2/environment.py

```diff
@@ -10,6 +10,7 @@
 """
 import os
 import sys
+from functools import reduce, partial
```

Comment:
Note that I moved this line while changing it above the `from jinja2 ..` imports for compliance with PEP-8, where corelib imports should go first.